crates/api/tests/issue809.rs

```diff
@@ -0,0 +1,33 @@
+use anyhow::{Context as _, Result};
```

Comment:
Nitty nit: could we call this file `invoke_func_via_table.rs` and the test `test_invoke_func_via_table` or something similar?

I think something along those lines would make this a proper test scenario rather than a specific regression test case (plus, this does a lot more than ensuring the module's signatures get registered).  Couldn't hurt to maybe have a comment to the GitHub issue inline as well.